packages/dds/merge-tree/src/test/testClientLogger.ts

```diff
@@ -46,14 +46,17 @@ function matchPropertiesHandleEmpty(a: PropertySet | undefined, b: PropertySet |
 	return matchProperties(a, b) || (arePropsEmpty(a) && arePropsEmpty(b));
 }
 
-type ClientMap = Partial<Record<"A" | "B" | "C" | "D" | "E", TestClient>>;
+type ClientMap<TClientName extends string> = Partial<Record<TClientName, TestClient>>;
```

Comment:
TS5 was clever enough to discover that some tests instantiate TestString with "id" or "fakeId", which is passed to createClientsAtInitialState.

This change permits arbitrary IDs, but preserves these IDs in the return Map (so if you pass "A", "B" the returned map has keys "A" and "B").

Alternatively, I could change "id" -> "A" and "fakeId" -> "B" if you prefer.

Reply:
I prefer this version, though I'm not a code owner for this area.